src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/tests/integration_test_blob_upload.py

```diff
@@ -0,0 +1,85 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+Integration test for blob upload
+
+The tests require environment variable 'test_connection_string' to set to the connection string
+of the storage account they will be run on.
+"""
+
+import os
+import os.path
+from unittest import TestCase
+from azure.cli.main import main as cli_main
+from .integration_test_base import StorageIntegrationTestBase
+
+
+class StorageBlobUploadPlainTests(TestCase):
+    def test_blob_upload_help(self):
+        with self.assertRaises(SystemExit):
+            cli_main('storage blob upload-batch -h'.split())
+
+
+class StorageBlobUploadIntegrationTests(StorageIntegrationTestBase):
+    def setUp(self):
+        self._keep_test_context = False
+        self._test_container_name = self.generate_new_container_name()
+        assert self._blob_service.create_container(self._test_container_name)
+
+    def tearDown(self):
+        if self._keep_test_context:
+            return
+
+        self._blob_service.delete_container(self._test_container_name)
+
+    def _get_test_resource_file(self, file_name):
+        result = os.path.join(self._resource_folder, file_name)
+        assert os.path.exists(result)
+
+        return result
+
+    def test_blob_upload_multiple_files_no_pattern(self):
+        url = 'http://{}/{}'.format(self._blob_service.primary_endpoint, self._test_container_name)
+        command = 'storage blob upload-batch -s {} -d {} --account-key {}'\
+            .format(self._resource_folder, url, self._test_account_key)
+
+        cli_main(command.split())
+
+        blobs = [b.name for b in self._blob_service.list_blobs(self._test_container_name)]
+        assert len(blobs) == 31
```

Comment:
Similar with the other tests.
If they are going to be skipped by Travis CI, when/how do we run them?